hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/DataChecksum.java

```diff
@@ -43,6 +49,9 @@
   public static final int CHECKSUM_CRC32C  = 2;
   public static final int CHECKSUM_DEFAULT = 3; 
   public static final int CHECKSUM_MIXED   = 4;
+
+  private static final Logger LOG = LoggerFactory.getLogger(DataChecksum.class);
+  private static volatile boolean useJava9Crc32C = Shell.isJavaVersionAtLeast(9);
```

Comment:
@steveloughran @dchuyko The flag `useJava9Crc32C` can be final and I'm thinking changing it final also avoids synchronization. What do you think?

Sorry for very late.

Reply:
@aajisaka The initial design was to have constructor method handle as static final in the main class and to not use the flag at all.
There was a concern that Java may not work as it must work: class or its constructor available by spec won't be found by standard means or there will be an error during invocation. I'd assume that in this case it would be fine to fail globally. Also some code around this stuff may be considered possible to fail.
These scenarios are described by try-catch blocks. In case something bad has happened on 9+ we switch the flag to use old internal plain java implementation and log the error.
The API should be thread safe/correct so we can imagine a race where multiple threads try the initialization, get errors and put down the flag.

Reply:
Thank you for the detailed comment. Sorry I missed the code that put downs the flag when get error. Now I'm +1.